### PR TITLE
Address https://github.com/Islandora-CLAW/CLAW/issues/302 ; Query Par…

### PR DESCRIPTION
**Pull request by ruebot**
…ameters: CHECKSUM (Optional - DEPRECATED)

See: https://github.com/Islandora-CLAW/CLAW/issues/302

### CODE REVIEW

**Comment by whikloj** [comment]
So this changes how the checksum is formatted. We used to just pass a checksum, but now that checksum is actually "&lt;checksum type&gt;=&lt;checksum&gt;".

For example, in the Chullo README there is an example of
```
$child_uri = $chullo->createResource(
        $uri,
        $rdf,
        ['Content-Type' => 'text/turtle'],
        $transaction,
        sha1($rdf)
    );
```
That now fails due to the change in Fedora, but using this change *also* doesn't work.

We need to change it to specify the checksum type as well as the checksum.

Like
```
$child_uri = $chullo->createResource(
        $uri,
        $rdf,
        ['Content-Type' => 'text/turtle'],
        $transaction,
        "sha1=" . sha1($rdf)
    );
```

So the code is fine, but the argument is changing.

I just want to make sure that we are okay with this, or should we separate the information out into two arguments or an array of arguments? &lt;shrug&gt;

**Comment by whikloj** [comment]
I haven't tried (but I will) using a "md5=blah" checksum

**Comment by whikloj** [comment]
Ok, so md5 seems to be disregarded.
I did
`$new_image = $api->createResource("", $image_content, $headers, "", 'md5=823478435');`
and it created the resource but
`$new_image = $api->createResource("", $image_content, $headers, "", 'sha1=823478435');`
gave a 409 Conflict response.
FYI, I made up the checksum.

File: src/Chullo.php [hunk 1]
```diff
@@ -226,21 +232,23 @@ public function saveResource(
     public function saveGraph(
         $uri,
         \EasyRdf_Graph $graph,
+        $checksum_algorithm = "",
```
Comment (whikloj) [inline src/Chullo.php]: You don't need this line.

File: src/Chullo.php [hunk 2]
```diff
@@ -226,21 +232,23 @@ public function saveResource(
...
 
         // Save it.
         return $this->saveResource(
             $uri,
             $turtle,
             ['Content-Type' => 'text/turtle'],
             $transaction,
-            $checksum
+            $checksum_value,
+            $checksum_algorithm
```
Comment (whikloj) [inline src/Chullo.php]: You also don't need this line, it will default to "sha1" which is what you are using.

**Comment by DiegoPino** [comment]
@ruebot that last one looks fine. $headers is the way, less arguments, closer to fcrepo REST API 👍

**Comment by whikloj** [comment]
Error in test is because you need to define the base_uri on the [Client here](https://github.com/Islandora-CLAW/chullo/pull/51/files#diff-ec5dd6e28cc50afa9c638d90b6b8eb6eR46). Check the other test.

**Pull request merged**